tests/saitests/py3/sai_qos_tests.py

```diff
@@ -787,6 +796,134 @@ def runTest(self):
+                                                     inner_frame=inner_pkt[scapy.IP])
+
+                    inner_pkt.ttl += 1
+
+                    exp_pkt = simple_tcp_packet(ip_src=inner_src_port_ip,
+                                                ip_dst=inner_dst_port_ip,
+                                                ip_dscp=exp_dscp,
+                                                ip_ecn=DEFAULT_ECN,
+                                                ip_ttl=DEFAULT_TTL)
+
+                    exp_pkt = Mask(exp_pkt)
+                    exp_pkt.set_do_not_care_scapy(Ether, 'src')
+                    exp_pkt.set_do_not_care_scapy(Ether, 'dst')
+                    exp_pkt.set_do_not_care_scapy(IP, 'id')
+                    exp_pkt.set_do_not_care_scapy(IP, 'ttl')
+                    exp_pkt.set_do_not_care_scapy(IP, 'chksum')
+
+                    send_packet(self, src_port_id, outer_pkt, DEFAULT_PKT_COUNT)
+
+                    try:
+                        port_index, _ = verify_packet_any_port(self, exp_pkt, ports=upstream_ptf_ports, timeout=3)
+                    except AssertionError:
+                        cause_for_failure[0] = True
+                        cause_for_failure[1].append("Expected packet with DSCP {} was not received ".format(dscp) +
+                                                    "on any of the ports: {}".format(upstream_ptf_ports))
+
+                    # validate pg counters increment by the correct pkt num
+                    time.sleep(1)
+                    pg_cntrs = sai_thrift_read_pg_counters(self.src_client, port_list['src'][src_port_id])
+
+                    for i in range(0, PG_NUM):
+                        try:
+                            if i == pg:
+                                assert (pg_cntrs[pg] == pg_cntrs_base[pg] + DEFAULT_PKT_COUNT)
+                                output_table.append("{}, {}, {}, PASS".format(pg, dscp, pg_cntrs))
+                            else:
+                                assert (pg_cntrs[i] == pg_cntrs_base[i])
+                        except Exception:
+                            cause_for_failure[0] = True
+                            cause_for_failure[1].append("PG counters are not incremented correctly for " +
+                                                        "priority group {} and dscp value {}".format(i, dscp))
+                            output_table.append("{}, {}, {}, FAIL".format(i, dscp, pg_cntrs))
+
+        finally:
+            headers = "Priority Group, DSCP, pg counters, Result"
+            curr_dir = os.getcwd()
+            with open(curr_dir + "/dscp_to_pg_mapping_ipip.txt", "w") as f:
+                f.write(headers+"\n")
+                f.write("\n".join(output_table))
+            if cause_for_failure[0]:
+                with open(curr_dir + "/dscp_to_pg_mapping_ipip_failures.txt", "w") as f:
+                    f.write("\n".join(cause_for_failure[1]))
+
+            print("END OF TEST")
+
+
```

Comment:
If there is some error happened, please fail the test.

Reply:
I am failing the test back on the dut - see `test_qos_sai`